external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaTimeUtils.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.spark.sql.kafka010
+
+import MockedSystemTime.currentMockSystemTime
+
+/**
+ * A class to get the current epoch time in millis.
+ * Having this class allows us the flexibility to use read epoch time or mocked epoch time
+ * for testing purposes so that we can advance epoch time in tests in a deterministic fashion
+ */
+trait Time {
+  def getCurrentTimeMillis: Long
+}
+
+/**
+ * To get the real epoch time
+ */
+class RealSystemTime extends Time {
+  override def getCurrentTimeMillis: Long = {
+    System.currentTimeMillis()
+  }
+}
+
+/**
+ * To return a mocked epoch time
+ */
+class MockedSystemTime extends Time{
+  override def getCurrentTimeMillis: Long = {
+    currentMockSystemTime
+  }
+}
+
+object MockedSystemTime {
+  var currentMockSystemTime = 0L
```

Comment:
The functionality is concise, it's great. It might be beneficial to expose a way to reset the time as well as get the time, but if you feel we can defer this unless required, that's OK.

Reply:
Ya let's add that functionality once it is actually needed.   I am not a fan of adding code that is not actually called